### PR TITLE
codesplitting reducers

### LINKED ISSUE #199: Codesplitting reducers
Right now we codesplit routes (see [`app/routes.js`](https://github.com/mxstbr/react-boilerplate/blob/v3.0.0/app/routes.js)) but all reducers are loaded with the main bundle – that doesn't make a lot of sense in large applications. They should be loaded when the corresponding route is visited.

*See [this StackOverflow answer](https://stackoverflow.com/questions/32968016/how-to-dynamically-load-reducers-for-code-splitting-in-a-redux-application) for an example implementation.*

----

> Note: If you want to submit a PR for this, work on the **v3.0.0 branch** please!

### PR DESCRIPTION
**Pull request by somus**
Fixes #199

### CODE REVIEW

File: app/containers/HomePage/reducer.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }),
+});
+
+function homeReducer(state = initialState, action) {
+  switch (action.type) {
+    case CHANGE_USERNAME:
+      // Delete prefixed '@' from the github username
+      return state
+        .setIn(['userData', 'username'], action.name.replace(/@/gi, ''));
+    case LOAD_REPOS:
```
Comment (mxstbr) [inline app/containers/HomePage/reducer.js]: Can we move the global `LOAD_REPOS`, `LOAD_REPOS_SUCCESS` and `LOAD_REPOS_ERROR` actions and constants back to the `App` component, only leaving the `CHANGE_USERNAME` in this reducer?

**Comment by mxstbr** [comment]
Thanks very much, just one small nitpick! Also, could you add the changes to the templates for the `npm run clean` command?

**Comment by somus** [comment]
When I build locally, eslint `react/prefer-stateless-function` error is not coming. It seems to appear only in Travis. Is that a newly added rule?

**Comment by mxstbr** [comment]
That's weird, lets disable it for now, I've added an issue to track this: #212

I'll do a functionality test now and merge right afterwards, thanks so much for your help @somus!

**Comment by mxstbr** [comment]
This works perfectly, thanks so much @somus!

**Pull request merged**